src/index.tsx

```diff
@@ -89,11 +94,15 @@ class LoadingBar extends Component<IProps, IState> {
         clearInterval(interval);
       }
     }, 1000);
+
+    this.setState({ progress: random, interval });
```

Comment:
This line here doesn't look okay to me since we only mutate the state inside the interval. And we add that to existing amount of the progress. By setting it like that you are resetting it to a random value